HugeCTR/include/hps/hash_map_backend_detail.hpp

```diff
@@ -91,6 +94,7 @@ namespace HugeCTR {
                                                                                              \
       /* Race-conditions here are deliberately ignored because insignificant in practice. */ \
       __VA_ARGS__;                                                                           \
+      while (payload.lck.load(std::memory_order_relaxed) != 0);                              \
       std::copy_n(payload.value, part.value_size, &values[(k - keys) * value_stride]);       \
```

Comment:
I think here is a logical error.

What if `payload.value` is rapidly changed. The `lck` doesn't guarantee that. During the `copy_n` payload value might change or become invalid.